core/variables.js

```diff
@@ -64,13 +64,13 @@ Blockly.Variables.allUsedVariables = function(root) {
   var variableHash = Object.create(null);
   // Iterate through every block and add each variable to the hash.
   for (var x = 0; x < blocks.length; x++) {
-    var blockVariables = blocks[x].getVars();
+    var blockVariables = blocks[x].getVarModels();
     if (blockVariables) {
       for (var y = 0; y < blockVariables.length; y++) {
-        var varName = blockVariables[y];
-        // Variable name may be null if the block is only half-built.
-        if (varName) {
-          variableHash[varName.toLowerCase()] = varName;
+        var variable = blockVariables[y];
+        // Variable ID may be null if the block is only half-built.
+        if (variable.getId()) {
+          variableHash[variable.name.toLowerCase()] = variable.name;
```

Comment:
Does it make sense for this function to return either a list of variable IDs or VariableModels instead of variable names since variables are no longer unique by name?